### PR TITLE
Shim helpers

### PR DESCRIPTION
**Pull request by rorticus**
Upgrade to use latest from shim, https://github.com/dojo/shim/pull/114 .

* Conversion to use `for..of`
* Using shimAmdDependencies and new intern loader (from grunt-dojo2)

### CODE REVIEW

**Comment by rorticus** [comment]
CI failures due to unreleased dependencies

File: src/List.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { forOf, Iterable, IterableIterator, ShimIterator } from '@dojo/shim/iterator';
+import {Iterable, IterableIterator, ShimIterator} from '@dojo/shim/iterator';
```
Comment (kitsonk) [inline src/List.ts]: minor nit... spaces between the braces

File: src/MultiMap.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import '@dojo/shim/Symbol';
 import Map from '@dojo/shim/Map';
-import { from as arrayFrom } from '@dojo/shim/array';
-import { forOf, Iterable, IterableIterator, ShimIterator } from '@dojo/shim/iterator';
+import {from as arrayFrom} from '@dojo/shim/array';
```
Comment (kitsonk) [inline src/MultiMap.ts]: spaces between braces, spaces between braces, spaces between braces (see it rhymes)

File: src/async/ExtensiblePromise.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { Iterable, forOf, isIterable, isArrayLike } from '@dojo/shim/iterator';
+import { Iterable, isIterable, isArrayLike } from '@dojo/shim/iterator';
```
Comment (kitsonk) [inline src/async/ExtensiblePromise.ts:1]: You see, you can do it! :-D

File: src/load/util.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
-import { forOf, isIterable, isArrayLike } from '@dojo/shim/iterator';
-import { Load } from '../load';
+import {isArrayLike, isIterable} from '@dojo/shim/iterator';
```
Comment (kitsonk) [inline src/load/util.ts]: 😭

File: src/request/Headers.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
-import { Headers as HeadersInterface } from './interfaces';
-import { IterableIterator, forOf, ShimIterator } from '@dojo/shim/iterator';
+import {Headers as HeadersInterface} from './interfaces';
```
Comment (kitsonk) [inline src/request/Headers.ts]: 🙈

File: src/request/providers/xhr.ts [hunk 6]
```diff
@@ -1,15 +1,14 @@
-import { Handle } from '@dojo/interfaces/core';
+import {Handle} from '@dojo/interfaces/core';
```
Comment (kitsonk) [inline src/request/providers/xhr.ts]: editors gone wild... next on the Discovery Channel

**Review by kitsonk — COMMENTED** [review]
Couple of thoughts...

File: src/List.ts (hunk 1 above)
Comment (rorticus) [reply src/List.ts]: **Ugh** WebStorm. I had to reinstall it and it looks like my special dojo2 style settings got disappeared. I'll have to go through the style guide and create a new one 😕

**Review by kitsonk — APPROVED** [review]

**Pull request merged**